testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksCloneTableCompatibility.java

```diff
@@ -0,0 +1,110 @@
+    public void testReadFromDeepClonedTable(boolean partitioned)
+    {
+        testReadClonedTable(false, partitioned);
+    }
+
+    private void testReadClonedTable(final boolean shallowClone, final boolean partitioned)
+    {
+        final String baseTable = "test_dl_base_table_" + randomNameSuffix();
+        final String clonedTableV1 = "test_dl_clone_tableV1_" + randomNameSuffix();
+        final String clonedTableV2 = "test_dl_clone_tableV2_" + randomNameSuffix();
+        try {
+            onDelta().executeQuery("CREATE TABLE default." + baseTable + " (a_number INT, b_string STRING) USING delta " +
+                    (partitioned ? "PARTITIONED BY (b_string) " : "") + "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + baseTable + "'");
+
+            onDelta().executeQuery("INSERT INTO default." + baseTable + " VALUES (1, \"a\")");
+
+            List<QueryAssert.Row> expectedRows = ImmutableList.of(row(1, "a"));
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + baseTable))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + baseTable))
+                    .containsOnly(expectedRows);
+
+            onDelta().executeQuery("INSERT INTO default." + baseTable + " VALUES (2, \"b\")");
+
+            onDelta().executeQuery("CREATE TABLE default." + clonedTableV1 + (shallowClone ? " SHALLOW CLONE" : " DEEP CLONE") + " default." + baseTable + " VERSION AS OF 1");
+
+            List<QueryAssert.Row> expectedRowsV1 = ImmutableList.of(row(1, "a"));
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + baseTable + " VERSION AS OF 1"))
+                    .containsOnly(expectedRowsV1);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + clonedTableV1))
+                    .containsOnly(expectedRowsV1);
+
+            onDelta().executeQuery("CREATE TABLE default." + clonedTableV2 + (shallowClone ? " SHALLOW CLONE" : " DEEP CLONE") + " default." + baseTable + " VERSION AS OF 2");
+
+            List<QueryAssert.Row> expectedRowsV2 = ImmutableList.of(row(1, "a"), row(2, "b"));
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + baseTable))
+                    .containsOnly(expectedRowsV2);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + baseTable))
+                    .containsOnly(expectedRowsV2);
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + baseTable + " VERSION AS OF 2"))
+                    .containsOnly(expectedRowsV2);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + clonedTableV2))
+                    .containsOnly(expectedRowsV2);
+
+            if (partitioned) {
+                List<QueryAssert.Row> expectedPartitionRows = ImmutableList.of(row("a"), row("b"));
```

Comment:
The below assertions aren't specific to partitioned or not. What's the motivation of them?

Reply:
I was intending to explicitly assert on partition values between base and cloned tables.